File: app/src/main/java/sgtmelon/scriptum/app/injection/component/PreferenceComponent.java
package sgtmelon.scriptum.app.injection.component;

import dagger.Component;
import sgtmelon.scriptum.app.injection.ArchScope;
import sgtmelon.scriptum.app.injection.module.DialogFindModule;
import sgtmelon.scriptum.app.injection.module.blank.PreferenceBlankModule;
import sgtmelon.scriptum.app.view.fragment.PreferenceFragment;

@ArchScope
@Component(modules = {
        PreferenceBlankModule.class,
        DialogFindModule.class
})
public interface PreferenceComponent {

    void inject(PreferenceFragment preferenceFragment);

}

File: app/src/main/java/sgtmelon/scriptum/presentation/dagger/component/PreferenceComponent.kt
package sgtmelon.scriptum.presentation.dagger.component

import dagger.BindsInstance
import dagger.Subcomponent
import sgtmelon.scriptum.presentation.dagger.ActivityScope
import sgtmelon.scriptum.presentation.dagger.module.base.InteractorModule
import sgtmelon.scriptum.presentation.dagger.module.base.ViewModelModule
import sgtmelon.scriptum.presentation.screen.ui.impl.preference.PreferenceFragment

/**
 * Component for [PreferenceFragment].
 */
@ActivityScope
@Subcomponent(modules = [InteractorModule::class, ViewModelModule::class])
interface PreferenceComponent {

    fun inject(fragment: PreferenceFragment)

    @Subcomponent.Builder
    interface Builder {
        @BindsInstance
        fun set(fragment: PreferenceFragment): Builder

        fun build(): PreferenceComponent
    }

}